curriculum/locales/english/learn-digital-signatures-by-building-a-wallet.md

```diff
@@ -4,206 +4,5928 @@
 
 ```js
-const stdout = await __helpers.getCommandOutput("node src/index.js");
-// console.log("Test 6: ", stdout);
-assert.include(stdout, "TOMTOM");
+await new Promise(res => setTimeout(res, 1000));
+const fileContents = await __helpers.getFile('learn-digital-signatures-by-building-a-wallet/generate-wallet.js');
+assert.match(fileContents, /const\s+publicKey\s*=\s*keyPair\s*\.\s*getPublic\s*\(\s*('|"|`)hex\1\s*\)\s*;?\s*$/);
 ```
 
 ### --seed--
 
-#### --"src/index.js"--
+#### --"generate-wallet.js"--
+
+```js
+import EC from 'elliptic';
 
-```rust
-const TOM = "TOMTOM";
+const ec = new EC.ec('p192');
+const keyPair = ec.genKeyPair();
 
 ```
 
-## 6
+## 12
+
+### --description--
+
+Create a `const privateKey` variable and use the `getPrivate` method on your keypair to get the private key in hex format, similar to how you got the public key.
+
+### --tests--
+
+You should have `const privateKey = keyPair.getPrivate('hex');` at the bottom of your `generate-wallet.js` file
+
+```js
+await new Promise(res => setTimeout(res, 1000));
+const fileContents = await __helpers.getFile('learn-digital-signatures-by-building-a-wallet/generate-wallet.js');
+assert.match(fileContents, /const\s+privateKey\s*=\s*keyPair\s*\.\s*getPrivate\s*\(\s*('|"|`)hex\1\s*\)\s*;?\s*$/);
+```
+
+### --seed--
+
+#### --"generate-wallet.js"--
+
+```js
+import EC from 'elliptic';
+
+const ec = new EC.ec('p192');
+const keyPair = ec.genKeyPair();
+const publicKey = keyPair.getPublic('hex');
+
+```
+
+## 13
+
+### --description--
+
+Now, you've extracted the two keys from the keypair. Below that, add a log to the console that prints `Public Key: <publicKey>` using a template literal.
+
+### --tests--
+
+You should have ``console.log(`Public Key: ${publicKey}`);`` at the bottom of your `generate-wallet.js` file
+
+```js
+await new Promise(res => setTimeout(res, 1000));
+const fileContents = await __helpers.getFile('learn-digital-signatures-by-building-a-wallet/generate-wallet.js');
+assert.match(fileContents, /console\s*\.\s*log\s*\(\s*`Public Key: \${\s*publicKey\s*}`\s*\)\s*;?\s*$/);
+```
+
+### --seed--
+
+#### --"generate-wallet.js"--
+
+```js
+import EC from 'elliptic';
+
+const ec = new EC.ec('p192');
+const keyPair = ec.genKeyPair();
+const publicKey = keyPair.getPublic('hex');
+const privateKey = keyPair.getPrivate('hex');
+
+```
+
+## 14
 
 ### --description--
 
-Run your code using:
+Add another log to print the private key in the same fashion so you can see what the keys look like.
+
+### --tests--
+
+You should have ``console.log(`Private Key: ${privateKey}`);`` at the bottom of your `generate-wallet.js` file
+
+```js
+await new Promise(res => setTimeout(res, 1000));
+const fileContents = await __helpers.getFile('learn-digital-signatures-by-building-a-wallet/generate-wallet.js');
+assert.match(fileContents, /console\s*\.\s*log\s*\(\s*`Private Key: \${\s*privateKey\s*}`\s*\)\s*;?\s*$/);
```

Comment:
The purpose is to try and make sure they stay on the right track. I usually mention to add the code at the bottom of the file, so I can check that there's nothing after it. Mainly, because I don't want them to add it above one of the existing lines. You're correct, but I think something like that is uncommon.